### ISSUE TITLE
Proposal for restructuring the templates and a custom project generator

### ISSUE THREAD

**Issue by JayaKrishnaNamburu** [teleporthq/teleport-code-generators#430]
Currently, our project templates are flat structured (In the UI for playground). So, if we can go ahead and adopt the two level of template management, we can start supporting more templates.

### Current flow
```
export
 -> templates
```

### Proposal
```
export
 -> templates
     -> react
       -> react-based-templates
    -> vue
      -> vue-based-templates
```

## Motivation

Community is moving fast and i see **snowpack** is also gaining traction. So, with this approach we can bring more templates under the generators. And even give options for the end users to select from multiple options instead of using the default one.

![proposal-templates](https://user-images.githubusercontent.com/11075561/74711921-1bc2d680-524b-11ea-81af-67728408b652.png)

## Custom Project Generator

We can even have a custom project generator, where user can input the template instead of we defaulting it. And add a `mapping.json` file so we can use that and map the paths in the generator. Like how we currently do project based attribute mapping.

Something like this
```js
const generator = createGenericProjectGenerator(STACK_CHOICE, STATIC_TEMPLATE, mapping)
```

**Comment by JayaKrishnaNamburu** [comment]
I have few ideas about the generic project generator approach, Bringing a life cycle approach for project generation. We can initialise some steps and each step receives all AST trees, template files, css styles sheets etc and the last stage is the ast to code conversion stage.

If we want to edit or change a file which we came across here --> https://github.com/teleporthq/teleport-code-generators/pull/399

We can allow the plugins to perform that, and the project generator waits for this plugin to respond back after changes. In this approach we can eliminate the project/template related bloat in the main project generators.

Here is the same kind of approach i came across in snowpack --> https://github.com/kevinpollet/pika-pack-plugin-template

**Comment by JayaKrishnaNamburu** [comment]
Fixed in #514 with custom project generator plugins.
